src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -170,9 +182,10 @@ public void ScheduleForExecution(AsyncTask asyncTask)
         }
 
         /// <summary>
-        ///     Flag determining whether or not the scheduler is operating in Test Mode.
+        /// Flag determining whether or not the scheduleed task will be
+        /// processed immediately or not. 
         /// </summary>
-        public bool IsTestMode { get; private set; }
+        public TaskProcessMode ProcessMode { get; set; }
```

Comment:
Can the `MutationTest` be made in such a way this property is not set *after* the scheduler is created? Having this flag switched halfway through execution isn't a good idea and there's no way for compiler to warn about it. It's better to have this as read-only property.